contracts/axelar-gateway/src/test_auth.rs

```diff
@@ -0,0 +1,545 @@
+
+    // should panic, epoch should return zero for unknown signer set
+    env.as_contract(&contract_id, || {
+        auth::validate_proof(&env, msg_hash, proof);
+    })
+}
+
+#[test]
+#[should_panic(expected = "failed ED25519 verification")]
+fn fail_validate_proof_invalid_signatures() {
+    let (env, contract_id, client) = setup_env();
+    let user = Address::generate(&env);
+
+    let signers = initialize(&env, &client, user, randint(0, 10), randint(1, 10));
+
+    let msg_hash = generate_random_payload_and_hash(&env);
+    let proof = generate_proof(&env, msg_hash.clone(), signers);
+
+    let different_msg = Bytes::from_array(&env, &[0x04, 0x05, 0x06]);
+    let different_msg_hash = env.crypto().keccak256(&different_msg).into();
+
+    // should panic, proof is for different message hash
+    env.as_contract(&contract_id, || {
+        auth::validate_proof(&env, different_msg_hash, proof);
+    })
+}
+
+#[test]
+#[should_panic(expected = "Error(Contract, #8)")]
+fn fail_validate_proof_empty_signatures() {
+    let (env, contract_id, client) = setup_env();
+    let user = Address::generate(&env);
+
+    let signers = initialize(&env, &client, user, randint(0, 10), randint(1, 10));
+
+    let msg_hash = generate_random_payload_and_hash(&env);
+    let mut proof = generate_proof(&env, msg_hash.clone(), signers);
+
+    // Modify signatures to make them invalid
+    let mut new_signers = Vec::new(&env);
+    for signer in proof.signers.iter() {
+        new_signers.push_back(ProofSigner {
+            signer: signer.signer,
+            signature: ProofSignature::Unsigned,
+        });
+    }
+    proof.signers = new_signers;
+
+    // validate_proof should panic, empty signatures
+    env.as_contract(&contract_id, || {
+        auth::validate_proof(&env, msg_hash, proof);
+    })
+}
+
+#[test]
+#[should_panic(expected = "Error(Contract, #6)")]
+fn fail_validate_proof_invalid_signer_set() {
+    let (env, contract_id, client) = setup_env();
+    let user = Address::generate(&env);
+
+    let signers = initialize(&env, &client, user, randint(0, 10), randint(1, 10));
+    let new_signers = generate_signers_set(&env, randint(1, 10), signers.domain_separator.clone());
+
+    let msg_hash = generate_random_payload_and_hash(&env);
+    let mut proof = generate_proof(&env, msg_hash.clone(), signers);
+
+    let new_proof = generate_proof(&env, msg_hash.clone(), new_signers);
+
+    proof.signers = new_proof.signers;
+
+    // validate_proof should panic, signatures do not match signers
+    env.as_contract(&contract_id, || {
+        auth::validate_proof(&env, msg_hash, proof);
+    })
+}
+
+#[test]
+#[should_panic(expected = "Error(Contract, #8)")]
+fn fail_validate_proof_threshold_not_met() {
+    let (env, contract_id, client) = setup_env();
+    let user = Address::generate(&env);
+
+    let signers = initialize(&env, &client, user, randint(0, 10), randint(1, 10));
+
+    let env = &env;
```

Comment:
This line seems a bit strange because there are only two places using `let env = &env;`. Could you look into this and remove it if possible?

Reply:
good point. it seems like it was just used to not have to write `&env` for a couple function calls which is confusing. removed both cases of this